src/Pact/Types/PactValue.hs

```diff
@@ -129,6 +130,14 @@ stripPactValueInfo = \case
   PGuard gu -> PGuard gu
   PModRef mr -> PModRef mr{_modRefInfo = def }
 
+stripAllPactValueInfo :: PactValue -> PactValue
+stripAllPactValueInfo = \case
```

Comment:
Note: `stripPactValueInfo` above isn't even enough, the author of that code did not cover `PGuard` which _can_ recursively contain pact values, the tests in this PR in `hash.repl` have a case with a user guard which does indeed fail if not stripped recursively.